patches/tModLoader/Terraria/DataStructures/ExtraJumpData.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace Terraria.DataStructures;
+
+/// <summary>
+/// A structure containing fields used to manage extra jumps
+/// </summary>
+public struct ExtraJumpData
+{
+	/// <summary>
+	/// Whether the extra jump can be used. This field is what should be set by equipment
+	/// </summary>
+	public bool Active;
+	/// <summary>
+	/// Whether the extra jump has been consumed
+	/// </summary>
+	public bool JumpAvailable;
```

Comment:
renamed `Active` to `Enabled`

`JumpAvailable` is mimicking the vanilla behaviour, which uses this flag for saying whether the jump can be used... aka it's "available" for consumption.  inverting the logic would be a headache for not much gain

`JumpAvailable` gets cleared immediately when the extra jumps starts (aside from contexts where all jumps are cleared)

unequipping an extra jump while `PerformingJump` is true just makes the visuals not appear.  re-equipping the jump while the "jump duration" is still in effect just makes the visuals appear again

Reply:
Needs to be included in documentation here. `Whether the extra jump has been consumed` implies that it's true if consumed. I don't think people will actually make that mistake, because the variable name is clear, but consider 
```
True if the jump has not yet been used. Will be set to false when a jump starts.
```
also:
```
Remember to also check Enabled
```

Reply:
noted.  i'll change the summary